src/FailureReceiver/FailureReceiverProvider.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace KaroIO\MessengerMonitorBundle\FailureReceiver;
+
+use KaroIO\MessengerMonitorBundle\Exception\FailureReceiverDoesNotExistException;
+use KaroIO\MessengerMonitorBundle\Exception\FailureReceiverNotListableException;
+use KaroIO\MessengerMonitorBundle\Locator\ReceiverLocator;
+use Symfony\Component\Messenger\Transport\Receiver\ListableReceiverInterface;
+
+class FailureReceiverProvider
```

Comment:
Make this `@internal`

Reply:
`@internal` + `@final` ? it's mocked somewhere

Reply:
Yep, both are fine - they mean subtly different things. The `@final` will actually prevent *us* from accidentally sub-classing it in the bundle :)